### ISSUE TITLE
Could not access property from Readonly<T> with index signature

### ISSUE THREAD

**Issue by yenshih** [Microsoft/TypeScript#13278]
**TypeScript Version:**  2.1.4

**Code**

```ts
type T = Readonly<{
    a: number
    [prop: string]: number
}>
const obj: T = { a: 0 }
obj.a // Property 'a' does not exist on type 'Readonly<{ [prop: string]: number; a: number; }>'.
```

**Expected behavior:**
No error.
**Actual behavior:**
Property 'a' does not exist on type 'Readonly<{ [prop: string]: number; a: number; }>'.

**Comment by ahejlsberg** [comment]
This was fixed by #12927 and works as expected in the nightly builds.

**Comment by yenshih** [comment]
Thanks a lot :)
